### PR TITLE
Migrate spyglass to AndroidX

### PR DESCRIPTION
**Pull request by sagar-waghmare-rsl**
- Migrated spyglass to AndroidX
- In build.gradle, replaced all occurrences of com.android.support 
  to its respective androidx components.
- In classes, replaced all android.support[.v7/v4] occurrences to its 
  respective androidx components.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -45,3 +45,7 @@ POM_SCM_DEV_CONNECTION=git@github.com:linkedin/Spyglass.git
 POM_LICENCE_NAME=The Apache Software License, Version 2.0
 POM_LICENCE_URL=http://www.apache.org/licenses/LICENSE-2.0.txt
 POM_LICENCE_DIST=repo
+
+#For AndroidX migration
+android.useAndroidX=true
+android.enableJetifier=true
```
Comment (csv8674xn) [inline gradle.properties]: Have we tested using this flag will make glide compatible with AndroidX? There is an issue here: https://github.com/bumptech/glide/issues/3080

**Comment by nhibner** [comment]
@sagar-waghmare-rsl - this is great work, thanks. If you would like to rebase (as we've made some changes to the build files recently) and address the above issue, I can take a look and try to get this merged in. I am currently planning to release an androidx-compatible version of Spyglass as version 2.0.

File: gradle.properties (hunk 1 above)
Comment (sagar-waghmare-rsl) [reply gradle.properties]: Hi @csv8674xn , yes we have tested this flag. Using this flag will make Glide compatible with AndroidX. We didn't encounter any compile time or runtime issues.

**Comment by nhibner** [comment]
@sagar-waghmare-rsl can you please squash your commits into a single commit? Other than that, looks good to me.

**Comment by sagar-waghmare-rsl** [comment]
> @sagar-waghmare-rsl can you please squash your commits into a single commit? Other than that, looks good to me.

@nhibner done. squash'ed all commits into one commit.

**Pull request merged**